docs/upgrade.html

```diff
@@ -45,6 +45,22 @@ <h5><a id="upgrade_240_notable" href="#upgrade_240_notable">Notable changes in 2
         The old overloaded functions are deprecated and we would recommend users to make their code changes to leverage the new methods (details
         can be found in <a href="https://cwiki.apache.org/confluence/display/KAFKA/KIP-520%3A+Add+overloaded+Consumer%23committed+for+batching+partitions">KIP-520</a>).
     </li>
+    <li>We are introducing incremental cooperative rebalancing to the clients' group protocol, which allows consumers to keep all of their assigned partitions during a rebalance
+        and at the end revoke only those which must be migrated to another consumer for overall cluster balance. To upgrade from eager to cooperative rebalancing, you must set all your clients
+        to use the new <code>CooperativeStickyAssignor</code>. This can be done with two rolling bounces: during the first one, add "cooperative-sticky" to the list of supported assignors
+        for each member (without removing the previous assignor -- note that if previously using the default, you must include that explicitly as well) before you bounce and/or upgrade it.
+        Once the entire group is on 2.4+ and all members have the <code>CooperativeStickyAssignor</code> among their supported assignors, perform a second rolling bounce and remove the
+        other assignor(s) so that by the end all members support only the cooperative protocol. For further details on the safe upgrade path or the cooperative rebalancing protocol,
+        see <a href="https://cwiki.apache.org/confluence/x/vAclBg">KIP-429</a>.
+    </li>
+    <li>There are some behavioral changes to the <code>ConsumerRebalanceListener</code>, as well as a new API. The <code>onPartitionsLost</code> method has been added to allow
+        users to react to abnormal circumstances where a consumer may have lost ownership of its partitions (such as a missed rebalance) and cannot commit offsets. By default, this will simply call
+        the existing <code>onPartitionsRevoked</code> API to align with previous behavior. Note however that <code>onPartitionsLost</code> will not be called when the set of lost partitions is empty.
+        This means that no callback will be invoked at the beginning of a rebalance by a new consumer joining the group. In addition, exceptions thrown during any of the listener's three callbacks
+        will no longer be swallowed, and will instead be re-thrown all the way up to the <code>Consumer.poll()</code> call. Note that if the consumer group is using the new cooperative rebalancing
+        described above, these callbacks will be invoked differently than when following the old eager rebalancing protocol. For details on the new callback semantics, see the
```

Comment:
`invoked differently than when following the old eager rebalancing protocol`: I'd suggest we either explicitly describe it here, or just drop this sentence and directly refer to the link. Normally statement like "there are other changes which we do not bother to mention here" would introduce more confusion than clarify.

Reply:
Fair point, will expand